File: LeetCode/Easy/0495-teemo-attacking/0495-teemo-attacking.java
class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        
        if(timeSeries.length == 0) return 0;
        if(timeSeries.length == 1) return duration;

        int ans = duration;
        int endTime = timeSeries[0] + duration;

        for(int i = 1; i < timeSeries.length; ++i) {
            int newTime = timeSeries[i] + duration;
            if(endTime < timeSeries[i]) {
                // 새로운 독에 걸림
                ans += duration;
                endTime = newTime;
            } else {
                // 이전 독 연장
                ans += newTime - endTime;
                endTime = newTime;
            }
        }


        return ans;
    }
}

File: LeetCode/Easy/0495-teemo-attacking/0495-teemo-attacking.kt
class Solution {
    fun findPoisonedDuration(timeSeries: IntArray, duration: Int): Int {
        if(timeSeries.size == 0) return 0
        if(timeSeries.size == 1) return duration

        var ans: Int = duration
        var end: Int = timeSeries[0] + duration

        for (i in 1 .. timeSeries.size - 1){

            val newEndTime = timeSeries[i] + duration

            if (end < timeSeries[i]) {
                // 새로 맞음
                ans += duration
                end = newEndTime
            } else {
                // 독 연장
                ans += newEndTime - end
                end = newEndTime
            }
        }

        return ans
    }
}